synapse/replication/resource.py

```diff
@@ -159,6 +159,17 @@ def replicate():
 
         result = yield self.notifier.wait_for_replication(replicate, timeout)
 
+        for stream_name, stream_content in result.items():
+            logger.info(
+                "Replicating %d rows of %s from %s -> %s",
+                len(stream_content["rows"]),
+                stream_name,
+                stream_content["position"],
+                request_streams.get(stream_name),
+            )
+            if stream_content["position"] == request_streams.get(stream_name):
```

Comment:
Hmm. Unfortunately this doesn't work because I was stupid and some of the response streams appear implicitly and don't have a corresponding entry in the request_streams.

For example asking for an ``"events"`` stream will also give an implicit ``"state_reset"`` stream.